### PR TITLE
docs(eos_cli_config_gen): Refactor Readme file for better readability

### PR DESCRIPTION
**Pull request by titom73**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

N/A

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Refactor the README file to be more readable and close to documentation generated by the role itself.

## How to test

Run development stack and connect to port 8000

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by titom73** [comment]
Need to change base branch to not target devel

**Review by carlbuchmann — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -7,90 +7,95 @@
   - [Role Inputs and Outputs](#role-inputs-and-outputs)
   - [Requirements](#requirements)
   - [Input Variables](#input-variables)
-    - [Terminal Settings](#terminal-settings)
+    - [Management](#management)
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Missing the following sections:
- Terminal Settings
- Load Interval
- Queue Monitor Length
- Queue Monitor Streaming
- Domain Lookup
- Speed-Group Settings
- MAC Address-table

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 2]
```diff
@@ -1316,29 +1154,95 @@ peer_filters:
         match: "< match as string >"
 ```
 
-### Router BGP Configuration
+### Routing
+
+### Service Routing Protocols Model
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: missing a "#" so it appears under section routing.

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
Please see comments!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 1 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Good catch. Fixed with commit 5951187

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 2 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Good catch. Fixed with commit 5951187

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 3]
```diff
@@ -137,289 +149,351 @@ Requirements are located here: [avd-requirements](../../README.md#Requirements)
...
-  width: < 0-32767 >
+management_interfaces:
+  < Management_interface_1 >:
+    description: < description >
+    shutdown: < true | false >
+    vrf: < vrf_name >
+    ip_address: < IPv4_address/Mask >
+    ipv6_enable: < true | false >
+    ipv6_address: < IPv6_address/Mask >
+    gateway: <IPv4 address of gateway>
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: I cannot see this being read in any template.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 4]
```diff
@@ -137,289 +149,351 @@ Requirements are located here: [avd-requirements](../../README.md#Requirements)
...
+    ipv6_gateway: <IPv6 address of gateway>
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: I cannot see this being read in any template.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
This helps a lot. We could consider sorting each level alphabetically so it is easy to know where to add new stuff. What do you think?
I found a two knobs not implemented in the templates, which I think we should remove from readme. I have not checked all knobs throughout. I stopped at aaa-authentication since I didn't find other discrepancies.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 4 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Remove it from documentation with commit 8c97a1a

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 3 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Remove it from documentation with commit 8c97a1a

**Comment by titom73** [comment]
> This helps a lot. We could consider sorting each level alphabetically so it is easy to know where to add new stuff. What do you think?
> I found a two knobs not implemented in the templates, which I think we should remove from readme. I have not checked all knobs throughout. I stopped at aaa-authentication since I didn't find other discrepancies.

Full documentation has been sorted alphabetically per section and also under each section with commit 3f03226

**Review by ClausHolbechArista — APPROVED** [review]
Looks **Great** To Me

**Comment by ClausHolbechArista** [comment]
Carl's requested changes have been done, so I will go ahead and merge

**Pull request merged**